### ISSUE TITLE
Synchronize with Boost.Multiprecision

### ISSUE THREAD

**Issue by nemothenoone** [NilFoundation/crypto3-multiprecision#3]
The synchronization with Boost.Multiprecision is required to handle low-priority updates and fixes with low effort amount.

**Comment by Zerg1996** [comment]
[Remove MPL usage](https://github.com/boostorg/multiprecision/commit/a9b4361c0aa290b8d0eb18f6b223dc98b6e692a1) wasn't complete (`boost::mpl_ -> std::integral_const<>`). Conversion errors for some types. Example: 
```
crypto3/libs/multiprecision/test/test_arithmetic.hpp:2874:4:
candidate function template not viable: no known conversion from 'typename is_twos_complement_integer<number<cpp_int_backend<500, 500, nil::crypto3::multiprecision::signed_magnitude, nil::crypto3::multiprecision::checked, void>, nil::crypto3::multiprecision::et_off>>::type' (aka 'mpl_::bool_<false>') to 'const std::integral_constant<bool, true>'
```

**Comment by Zerg1996** [comment]
Fixed in #8
